### PR TITLE
Implement function calls via table lookup and remove explicit specifications

### PR DESCRIPTION
**Pull request by jimidle**
Replaces tons of explicit specifications of generic function calls with a a table of standard functions. We create a visitor for function call building to go with this. 

Closes: #339
Progresses: #275

### CODE REVIEW

File: core/src/main/scala/com/databricks/labs/remorph/parsers/intermediate/extensions.scala [hunk 1]
```diff
@@ -138,3 +138,5 @@ case class RenameConstraint(oldName: String, newName: String) extends TableAlter
 case class RenameColumn(oldName: String, newName: String) extends TableAlteration
 
 case class AlterTableCommand(tableName: String, alterations: Seq[TableAlteration]) extends Catalog {}
+
+case class ExpressionList(expressions: Seq[Expression]) extends Expression {}
```
Comment (sundarshankar89) [inline core/src/main/scala/com/databricks/labs/remorph/parsers/intermediate/extensions.scala:142]: why do we need this?

File: core/src/main/scala/com/databricks/labs/remorph/parsers/tsql/TSqlFunctionBuilder.scala [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+    "PERMISSIONS" -> FunDef(0, 2),
+    "PWDENCRYPT" -> FunDef(1, 1),
+    "PWDCOMPARE" -> FunDef(2, 3),
+    "SESSIONPROPERTY" -> FunDef(1, 1),
+    "SUSER_ID" -> FunDef(0, 1),
+    "SUSER_NAME" -> FunDef(0, 1),
+    "SUSER_SID" -> FunDef(0, 2),
+    "SUSER_SNAME" -> FunDef(0, 1),
+    "USER_ID" -> FunDef(0, 1),
+    "USER_NAME" -> FunDef(0, 1))
```
Comment (sundarshankar89) [inline core/src/main/scala/com/databricks/labs/remorph/parsers/tsql/TSqlFunctionBuilder.scala:201]: should we have an IR for holding un-covered expression so we can transpile as is on target side.

File: core/src/main/scala/com/databricks/labs/remorph/parsers/intermediate/extensions.scala (hunk 1 above)
Comment (jimidle) [reply core/src/main/scala/com/databricks/labs/remorph/parsers/intermediate/extensions.scala:142]: It has gone. I changed the grammar but that is in another PR. I will rebase when that goes through

File: core/src/main/scala/com/databricks/labs/remorph/parsers/tsql/TSqlFunctionBuilder.scala (hunk 2 above)
Comment (jimidle) [reply core/src/main/scala/com/databricks/labs/remorph/parsers/tsql/TSqlFunctionBuilder.scala:201]: > should we have an IR for holding un-covered expression so we can transpile as is on target side.

I am going to close this PR and resubmit it without the fork. 

On functions that are not covered - we may also want to collect those as lint messages (see other PR about collecting messages), but also it needs to check for user defined functions, but that is functionality for another PR I think.

**Comment by jimidle** [comment]
Closing to resubmit without fork

**Pull request closed without merging**